main.py

```diff
@@ -0,0 +1,146 @@
+            print("Invalid customer ID")
+            return withdrawal(login())
+
+
+def deposit(customer_id: int) -> None:
+    """
+    This function asks the user for the amount he wishes to deposit
+    param: customer_id: the customer ID of the user who wishes to deposit money
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:
+        customers_data = json.load(customers_file)
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                amount = int(input("Enter the amount you wish to deposit"))
+                customer["balance"] += amount
+                print(f"Your new balance is {customer['balance']}")
+                with open("customers.json", 'w') as customers_file_new:
+                    json.dump(customers_data, customers_file_new, indent=4)
+                return
+        else:
+            print("Invalid customer ID")
+            return deposit(login())
+
+
+def change_password(customer_id: int) -> None:
+    """
+    This function asks the user for his new password and changes it
+    param: customer_id: the customer ID of the user who wishes to change his password
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:
+        customers_data = json.load(customers_file)
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                password = int(input("Enter your new password"))
+                customer["password"] = password
+                print("Your password has been changed")
+                with open("customers.json", 'w') as customers_file_new:
+                    json.dump(customers_data, customers_file_new, indent=4)
+                return
+        else:
+            print("Invalid customer ID")
+            return change_password(login())
+
+
+def balance(customer_id: int) -> None:
+    """
+    This function prints the balance of the user
+    param: customer_id: the customer ID of the user who wishes to see his balance
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:
+        customers_data = json.load(customers_file)
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                print(f"Your balance is {customer['balance']}")
+                return
+        else:
+            print("Invalid customer ID")
+            return balance(login())
+
+
+def main() -> None:
+    """
+    This function is the main function of the program, it calls the other functions
+    return: None
+    """
+    customer_id = login()
+    print("Welcome to the ATM")
+    while True:
+        print("What would you like to do?")
+        print(" 1. Withdrawal")

```

Comment:
Maybe an overkill for this assignment, but you can define a dictionary where the key is the option number and the value is its matching function and the name of the action, and use if to fully operate the ATM's actions (and to print the options). This way you only need to add a new action in one place